src/rgw/driver/sfs/bucket.cc

```diff
@@ -238,12 +256,10 @@ bool SFSBucket::is_owner(User* user) {
 int SFSBucket::check_empty(const DoutPrefixProvider* dpp, optional_yield y) {
```

Comment:
Yup, I wonder if there are any `s3 tests` for this.

Reply:
Surprising to me check_empty is only called from RGWDeleteBucket::execute before doing the remove_bucket call. Surprising because I would have expected that to be part of remove_bucket

Reply:
The answer here needs to be, I think, that the multipart uploads are simply aborted if the bucket is removed. The multipart upload is a promise of eventually creating something in that bucket, and as such I don't think it's reasonable to hold the bucket to exist just because someone forgot to complete or abort a multipart upload.

Reply:
Good to know, thanks!